homework-submission/App.js

```diff
@@ -1,105 +1,161 @@
 const path = '/data.json';
 
-const ListItem = ({ id, description, deadline, className }) => {
+// to add list item with classes and onClick event
+//props are description, deadline, onClick, className
+
+const ListItem = ({ description, deadline, onClick, className }) => {
   return (
-    <li className={className}>
-      Todo List Number: {id} Description: {description} Deadline: {deadline}
+    <li className={className} onClick={onClick}>
+      Description: {description} Deadline: {deadline}
     </li>
   );
 };
 
-class TodoApp extends React.Component {
+//this list takes list items and loops through data to fill list items
+
+const List = ({ todoList }) => {
+  return (
+    <ul>
+      {todoList.map(({ id, description, deadline, done }) => (
+        <ListItem
+          description={description}
+          deadline={deadline}
+          className={done === true ? 'strike_through' : ''}
+          onClick={event => classChanger(event)}
+          key={id}
+        />
+      ))}
+    </ul>
+  );
+};
+
+//input element is used in form
+// probs are type, name, placeholder, onChange
+
+const Input = ({ type, name, placeholder, onChange }) => {
+  return <input type={type} name={name} placeholder={placeholder} onChange={onChange} />;
+};
+
+//button element is used in form
+// probs are value, onClick
+
+const Button = ({ value, onClick }) => {
+  return <button onClick={onClick}> {value} </button>;
+};
+
+// this todos class will be populated by list items, inputs and buttons
+// the state of this class will be used to update the list
+
+class Todos extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
       todos: [],
     };
   }
-
+  // to add initial todo list
   componentDidMount() {
-    readFile(path).then(response => {
-      this.setState(() => ({
-        todos: response,
-      }));
+    readFile(path)
+      .then(data => data.json())
+      .then(response => {
+        this.setState(() => ({
+          todos: response,
+        }));
+      })
+      .catch(error => {
+        this.setState(() => {
+          todos: error.message;
+        });
+      });
+  }
+  //to add new todos with unique id's. Even though a todo is deleted, none of the id's will be coincided.
+  addTodo(event, NewTodo, todos) {
+    event.preventDefault();
+    const lastTodoIndex = todos.length - 1;
+    NewTodo.id = todos[`${lastTodoIndex}`].id + 1;
+    todos.push(NewTodo);
+    this.setState({
+      todos: todos,
     });
   }
-
+  // to delete todos. User must insert todos place in the list to delete it
+  deleteTodo(event, TodoNumber, todos) {
+    event.preventDefault();
+    const updatedList = todos.filter((todo, index) => {
+      return index != TodoNumber - 1;
+    });
+    this.setState({ todos: updatedList });
+  }
   render() {
     const todos = this.state.todos;
+    // this new todo is produced to be pushed when Add new todo button is clicked.
     const NewTodo = {
       id: 0,
```

Comment:
State's existence for handling this job for us, so why not using it, it will make it a lot easier and familiar for others